scripts/modules/b2b-account/email-quote.js

```diff
@@ -23,71 +23,77 @@ define([
-                if(!accountInfo.isSalesRep){
-                $("#sendEmailText").val(accountInfo.email);
+                if (!accountInfo.isSalesRep) {
+                    $("#sendEmailText").val(accountInfo.email);
                 }
                 $(".modal-dialog").attr("id", "mz-email-modal");
 
-            $('[data-mz-action="handleSendemail"]').click(function(e) {
-                e.preventDefault();
-                var invalidEmails=[];
-                var validEmails=[];
-                var emails; 
-               if(!self.model.attributes.inputEmails && !accountInfo.isSalesRep){
-               emails=$("#sendEmailText").val();
-               }
-               else{
-               emails=self.model.attributes.inputEmails;
-               }
-
-                var result = emails.split(",");
-                for(var i = 0;i < result.length && i < maxEmailAllowed;i++){
-                    if(!result[i].match(Backbone.Validation.patterns.email)){
-                        invalidEmails.push(result[i]);
-                    }else{
-                        validEmails.push(result[i]);
+                $('[data-mz-action="handleSendemail"]').click(function (e) {
+                    e.preventDefault();
+                    var invalidEmails = [];
+                    var validEmails = [];
+                    var emails;
+                    if (!self.model.attributes.inputEmails && !accountInfo.isSalesRep) {
+                        emails = $("#sendEmailText").val();
+                    }
+                    else {
+                        emails = self.model.attributes.inputEmails;
                     }
-                }
 
-                if(invalidEmails.length===0){
-                var quote = new QuoteModels.Quote({
-                    id: self.model.get("accountId")
-                });
-                return quote.apiModel.email(validEmails).then(function (res) {
-                   // to do after email sent successfully
-                    // emailAddress:validEmails
+                    var result = emails.split(",");
+                    for (var i = 0; i < result.length && i < maxEmailAllowed; i++) {
+                        if (!result[i].match(Backbone.Validation.patterns.email)) {
+                            invalidEmails.push(result[i].trim());
+                        } else {
+                            validEmails.push(result[i].trim());
+                        }
+                    }
+                    var json = {
+                        "emailIds": validEmails,
+                        "id": self.model.get("accountId")
+                    };
 
-                }, function (error) {
-                    if (error.message !== undefined){
-                        self.showMessageBar(error);
-                    }else{
-                        self.showMessageBar({
-                            message: "Bad Request Error"
+                    var quoteId = self.model.get("accountId");
```

Comment:
why are you setting `accountId` as `quoteId`

Reply:
Yes that's bad variable name I had used.I  have changed it to quoteId.